components/script/dom/bindings/str.rs

```diff
@@ -113,6 +113,63 @@ pub fn is_token(s: &[u8]) -> bool {
     })
 }
 
+/// Returns whether the language is matched, as defined by
+/// [RFC 4647](https://tools.ietf.org/html/rfc4647#section-3.3.2).
+pub fn extended_filtering(tag: &str, range: &str) -> bool {
+    let lang_ranges: Vec<&str> = range.split(',').collect();
+
+    lang_ranges.iter().any(|&lang_range| {
+        // step 1
+        let range_subtags: Vec<&str> = lang_range.split('\x2d').collect();
+        let tag_subtags: Vec<&str> = tag.split('\x2d').collect();
+        if tag.eq_ignore_ascii_case(lang_range) || lang_range.eq_ignore_ascii_case("*") {
+            return true;
```

Comment:
This check is unnecessary here. The spec at this point only defines what it means to have a match, it doesn't intend for you to actually make a match here.